src/FFmpegVideoEncoder.cs

```diff
@@ -175,26 +204,46 @@ private void InitialiseEncoder(AVCodecID codecID, int width, int height, int fps
 
                 if (_codecID == AVCodecID.AV_CODEC_ID_H264)
                 {
-                    //_videoCodecContext->profile = ffmpeg.FF_PROFILE_H264_CONSTRAINED_BASELINE;
-                    ffmpeg.av_opt_set(_encoderContext->priv_data, "profile", "baseline", 0).ThrowExceptionIfError();
-                    //ffmpeg.av_opt_set(_videoCodecContext->priv_data, "packetization-mode", "0", 0).ThrowExceptionIfError();
-                    //ffmpeg.av_opt_set(_pCodecContext->priv_data, "preset", "veryslow", 0);
-                    //ffmpeg.av_opt_set(_videoCodecContext->priv_data, "profile-level-id", "42e01f", 0);
+                    var parameters = Helper.ParseWebRtcParameters(Format.Parameters);
+                    bool configured = false;
+                    if (parameters.TryGetValue("profile-level-id", out string? profileLevelId))
+                    {
+                        if (profileLevelId!=null && profileLevelId.Length == 6)
+                        {
+                            string profileHex = profileLevelId.Substring(0, 2);
+                            string levelHex = profileLevelId.Substring(4, 2);
+
+                            string profile = Helper.GetH264ProfileName(profileHex);
+                            string level = Helper.GetH264LevelValue(levelHex);
 
-                    ffmpeg.av_opt_set(_encoderContext->priv_data, "tune", "zerolatency", 0).ThrowExceptionIfError();
+                            if (!string.IsNullOrEmpty(profile) && !string.IsNullOrEmpty(level))
+                            {
+                                av_opt_set(_encoderContext->priv_data, "profile", profile, AV_OPT_SEARCH_CHILDREN);
+                                av_opt_set(_encoderContext->priv_data, "level", level, AV_OPT_SEARCH_CHILDREN);
+                                configured = true;
+                            }
+                        }
+
+                    }
+                    if (!configured)
+                    {
+                        av_opt_set(_encoderContext->priv_data, "profile", "baseline", AV_OPT_SEARCH_CHILDREN);
+                    }
+                    av_opt_set(_encoderContext->priv_data, "tune", "zerolatency", 0);
                 }
                 else if ((_codecID == AVCodecID.AV_CODEC_ID_VP8) || (_codecID == AVCodecID.AV_CODEC_ID_VP9))
                 {
-                    ffmpeg.av_opt_set(_encoderContext->priv_data, "quality", "realtime", 0).ThrowExceptionIfError();
+                    av_opt_set(_encoderContext->priv_data, "quality", "realtime", 0).ThrowExceptionIfError();
+                    CalculateVp8BitrateParameters(width, height, fps, 0.1);

```

Comment:
I've had not-so-good experiences with this concept of processing things in initialization, it may cause a sense of lag in the video while switching encoder specifications.

You may want to put the processing in a separate method and set the intrinsics for the encoder initialization, with necessary conditionals and private fields.